core/gethwrappers/go_generate_test.go

```diff
@@ -102,9 +103,17 @@ func compareCurrentCompilerArtifactAgainstRecordsAndSoliditySources(
 and commit the changes`, versionInfo.AbiPath, versionInfo.BinaryPath, recompileCommand))
 }
 
+func TestMain(m *testing.M) {
```

Comment:
I think that is not necessary since the inits will execute before this `TestMain` is called, and they are not problematic since they do not depend on flags. `testing.Short` is a just a quirk that requires flags to be parsed already.

Regardless, that rootDir var is a little strange. Doesn't seem like it needs to be an init at all? It is just supporting `TestCheckContractHashesFromLastGoGenerate` :thinking:

Reply:
Good, can we just move it to this test then and get rid of the init function at all?